tests/test.ml

```diff
@@ -27,11 +27,80 @@ let test_metrics () =
     "
     output
 
+
+(* "^[a-zA-Z_][a-zA-Z0-9_]*$" *)
+let valid_labels = [
+  "_";
+  "a";
+  "aA0b1B9c8C7z6Z5y4Y3x2X1";
+  "_______";
+]
+
+let invalid_labels = [
+  "";
+  "1";
+  "a bad label";
+]
+
+let check_valid_label label () =
+  let _l: LabelName.t = LabelName.v label in
+  ()
+
+let check_invalid_label label () =
+  try
+    let _l: LabelName.t = LabelName.v label in
+    failwith (label ^ " should be an invalid label")
+  with _ -> ()
```

Comment:
I think this is just going to catch the exception you throw on the line above...